libkernelflinger/vars.c

```diff
@@ -254,32 +254,72 @@ static void set_provisioning_mode(BOOLEAN provisioning)
 	current_state = provisioning ? UNLOCKED : LOCKED;
 }
 
-enum device_state get_current_state()
+static EFI_STATUS read_efi_device_state(UINT8 *state)
 {
-	UINT8 *stored_state;
-	UINTN dsize;
 	EFI_STATUS ret;
+	UINT8 *stored_state = NULL;
+	UINTN dsize;
 	UINT32 flags;
+
+	ret = get_efi_variable((EFI_GUID *)&fastboot_guid, OEM_LOCK,
+				       &dsize, (void **)&stored_state, &flags);
+	if (EFI_ERROR(ret)) {
+		if (ret == EFI_NOT_FOUND)
+			debug(L"Can not find device state EFI variable");
+		else
+			efi_perror(ret, L"Failed to read device state from EFI variable");
+		return ret;
+	}
+	if (!dsize || !stored_state) {
+		error(L"Read device state from EFI variable, but data size is 0");
+		ret = EFI_COMPROMISED_DATA;
+		goto out;
+	}
+	if (flags & EFI_VARIABLE_RUNTIME_ACCESS) {
+		error(L"Read device state from EFI variable, but it can be accessed in runtime!");
+		ret = EFI_SECURITY_VIOLATION;
+		goto out;
+	}
+
+	*state = *stored_state;
+	debug(L"Success read device state from EFI variable, state: %d", *state);
+out:
+	if (stored_state)
```

Comment:
If you arrive there `stored_state` cannot be `NULL`

Reply:
if (flags & EFI_VARIABLE_RUNTIME_ACCESS)